experimental/InjectiveResolutions/src/LocalCohomology.jl

```diff
@@ -0,0 +1,639 @@
+    push!(
+      H,
+      SectorPartitionLC(M, j, I.ideal, Hj, maps_needed(kQ, Hj)),
+    )
+  end
+  return H
+end
+
+# given a matrix with entries in a MPolyDecRing or a MPolyQuoRing, return the matrix with the corresponding scalar coefficients
+@doc raw"""
+  get_scalar_matrix(A::MonoidAlgebra, M)
+
+Given a monomial matrix, return the corresponding coefficient matrix. 
+"""
+function get_scalar_matrix(A::MonoidAlgebra, M)
+  k = coefficient_ring(A)
+  return map(x -> evaluate(x,ones(elem_type(k),ngens(A))), M)
+end
+
+# given a monoid algebra k[Q] compute linear functions \tau_1,...,\tau_n that define Q. 
+# there is one defining linear functional for every facet of Q and Q is the intersection of the halfspaces {\tau_i \leq 0} 
+
+@doc raw"""
+  compute_taus(kQ::MonoidAlgebra, J::IndecInj...)
+
+This function computes a vector $\tau^j \in (\mathbb{Z}\cup \infty)^n$ for every indecomposable injective $J^j$.
+See [HM05](@cite) (Section 6).
+"""
+function compute_taus(kQ::MonoidAlgebra, J::IndecInj...)
+  # get linear functionals tau_i for every hyperplane bounding Q, i.e. for every facet
+  A, _ = halfspace_matrix_pair(facets(kQ.cone))
+  H = []
+  for i in 1:length(kQ.hyperplanes)
+    push!(H, [kQ.hyperplanes[i].hyperplane, A[i, :]])
+  end
+
+  # for every indecomposable injectives J_j = k{a_j + F_j - Q} compute the vector \tau^j used in Section 6 of HM05  
+  _tau = []
+  for J_i in J
+    in
+    tau_i = []
+    for h in H
+      if issubset(J_i.face.poly, h[1])
+        a_ih = dot(h[2], J_i.vector)
+        push!(tau_i, a_ih)
+      else
+        push!(tau_i, -Inf)
+      end
+    end
+    push!(_tau, tau_i)
+  end
+
+  #transform _tau consisting of length(kQ.hyperplanes)-lists into a list of length(J) (divide by coordinate index)
+  tau = []
+  for i in 1:length(kQ.hyperplanes)
+    push!(tau, [t[i] for t in _tau])
+  end
+  return tau
+end
+
+@doc raw"""
+  get_halfspace_eq(kQ::MonoidAlgebra)
+
+Returns a finite set of positive halfspaces such that $Q$ is the intersection of them. 
+
+!!! note 
+The corresponding semigroup $Q$ must be saturated. 
+"""
+function get_halfspace_eq(kQ::MonoidAlgebra)
+  A, _ = halfspace_matrix_pair(facets(kQ.cone))
+  H = []
+  for i in 1:length(kQ.hyperplanes)
+    push!(H, [kQ.hyperplanes[i].hyperplane, A[i, :]])
+  end
+  return H
+end
+
+@doc raw"""
+  sector_partition(kQ::MonoidAlgebra,phi::Union{Vector{Any},MatElem{T}},psi::Union{Vector{Any},MatElem{T}},j::Integer,k::Integer,J::IndecInj...,) where {T<:FieldElem}
+
+Given a cochain complex of injective modules J^0 -> J^1 -> J^2, where phi: J^0 -> J^1 and psi: J^1 -> J^2, let J = J^0 + J^1 + J^2
+and j = |J^0|, k = |J^0| + |J^1|. This function computes a sector partition of the local cohomology module ker(psi)/im(phi). 
+"""
+function sector_partition(
+  kQ::MonoidAlgebra,
+  phi::Union{Vector{Any},MatElem{<:T}},
+  psi::Union{Vector{Any},MatElem{<:T}},
+  j::Integer,
+  k::Integer,
+  J::IndecInj...,
+) where {T<:FieldElem}
```

Comment:
@HechtiDerLachs the failure is due to the following: If phi and psi are both of type Vector{Any}, then there is nothing restricting the value of the type variable T.
This is usually a symptom of some bad code design somewhere. Without looking any deeper into the code in this PR (I could do that tomorrow if still required), I am confused why phi and psi can be a matrix (sounds reasonable for the internal representation of a morphism) *or a Vector{Any}*. Should the *Vector{Any}* encode some special case like zero or identity? If yes, why not use such a matrix instead.
If you can point me to the line of code where such a Vector{Any} is created, I may have a suggestion how to change it to something that doesn't complain

Reply:
Thanks! I think, we fixed it.